src/sql/workbench/services/notebook/common/notebookService.ts

```diff
@@ -157,6 +157,8 @@ export interface INotebookEditor {
 export interface INavigationProvider {
 	providerId: string;
 	hasNavigation: boolean;
+	hasNext(uri: URI): Promise<boolean>;
+	hasPrevious(uri: URI): Promise<boolean>;
 	getNavigation(uri: URI): Thenable<azdata.nb.NavigationResult>;
```

Comment:
So, isn't the basic information already embedded in the NavigationResult? Do we need these extra methods, or could we just wait on the result of getNavigation and then do everything from there?